agpypeline/geometries.py

```diff
@@ -149,7 +155,10 @@ def polygon_from_ring(ring: ogr.Geometry, epsg: int = None) -> Optional[ogr.Geom
     if epsg is not None:
         if ref_sys.ImportFromEPSG(int(epsg)) == ogr.OGRERR_NONE:
             poly.AssignSpatialReference(ref_sys)
+        elif filename is not None:
```

Comment:
This may be better handled by the calling application/function since it may not be an error that the polygon can't be created; the call may not involve a file; the calling context may also be different enough that the message is misleading (I can't think of an example at this time). I have found in general that it's better to have the calling function handle an error when something happens in a low level function like this.